### DISCUSSION TITLE
Order of `beforeResolve` vs `beforeMount`

### DISCUSSION THREAD

**Discussion by MetRonnie** [vuejs/router#1710] [Help and Questions] [answered]
I'm upgrading an app from Vue 2 to 3 and came across a problem due to an apparent change in the order of the router `beforeResolve()` global navigation guard and the component `beforeMount()` lifecycle hook.

In vue 2 and vue-router 3, the order was:
1. `router.beforeEach()`
1. `router.beforeResolve()`
1. `component.beforeMount()`

Whereas in vue 3 and vue-router 4 the order is:
1. `router.beforeEach()`
1. `component.beforeMount()`
1. `router.beforeResolve()`

(I have determined this order by adding breakpoints in each.)

This change does not seem to be documented in the upgrade guide, or at least I couldn't find it anywhere.

**Comment by posva** [comment, marked as answer]
This is because Routing Lifecycle and Component Lifecycle are independent. Callbacks passed to hooks should therefore also be independent and not rely on the order.
